packages/financial-templates-lib/test/truffle/UniswapPriceFeed.js

```diff
@@ -301,43 +312,61 @@ contract("UniswapPriceFeed.js", function(accounts) {
     assert.equal(uniswapPriceFeed.getLastBlockPrice().toString(), toWei("2"));
   });
 
-  describe("Can return non-18 precision prices", function() {
+  describe("Can handle non-18 precision pool prices and return non-18 precision prices", function() {
     let scaleDownPriceFeed, scaleUpPriceFeed;
+    let quoteCurrencyPrecision = 8;
+    let baseCurrencyPrecision = 6;
+    const convertQuoteCurrency = Convert(quoteCurrencyPrecision);
+    const convertBaseCurrency = Convert(baseCurrencyPrecision);
+
     beforeEach(async function() {
-      // Here we specify that the pool is reporting a 6 decimal precision price, and we want prices in
+      // The UniswapPriceFeed's _getPriceFromSyncEvent will return prices in same precision as
+      // the base currency. But we also change the quote currency precision to test that the
+      // UniswapPriceFeed correctly handles it.
+      let quoteToken = await Token.new("Uni Token", "UQ", quoteCurrencyPrecision, { from: owner });
+      let baseToken = await Token.new("Uni Token", "UB", baseCurrencyPrecision, { from: owner });
+
+      // Not inverted so quote token is token0
+      await uniswapMock.setTokens(quoteToken.address, baseToken.address);
+
+      // Here we specify that the pool is returning a 6 decimal precision price, and we want prices in
```

Comment:
Nit: The pool is returning a 6 decimal precision price, but it doesn't look like we actually specify it here anymore; it's pulled from the pool, right? The part where it's specified in the function is deleted. Same nit for `scaleDownPriceFeed`.

Reply:
Correct the price is pulled from one of the Uniswap pair's constituent tokens. The `uniswapMock.setTokens()` method sets the tokens so I'll make this more clear.

Reply:
Removed all references to `base` and `quote` from the tests and also made it clear why we're expecting the pool to return 6 precision and how we convert that either into 12 or 4